### PR TITLE
chore: update dependencies for RTL

### PR DESCRIPTION
**Pull request by web-padawan**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -29,28 +29,25 @@
   "dependencies": {
     "polymer": "^2.0.0",
     "vaadin-themable-mixin": "vaadin/vaadin-themable-mixin#^1.2.0",
```
Comment (yuriy-fix) [inline bower.json:31]: Can be bumped as well.

File: bower.json [hunk 2]
```diff
@@ -29,28 +29,25 @@
...
     "iron-test-helpers": "^2.0.0",
     "webcomponentsjs": "^1.0.0",
     "web-component-tester": "^6.1.5",
-    "vaadin-combo-box": "vaadin/vaadin-combo-box#^4.2.1",
-    "vaadin-demo-helpers": "vaadin/vaadin-demo-helpers#^3.0.0",
-    "vaadin-date-picker": "vaadin/vaadin-date-picker#^4.0.0",
+    "vaadin-combo-box": "vaadin/vaadin-combo-box#^5.1.0-alpha1",
+    "vaadin-demo-helpers": "vaadin/vaadin-demo-helpers#^3.1.0-alpha1",
+    "vaadin-date-picker": "vaadin/vaadin-date-picker#^4.1.0-alpha1",
     "vaadin-dialog": "vaadin/vaadin-dialog#^2.2.1"
```
Comment (yuriy-fix) [inline bower.json:51]: Nit: not a blocker. We can wait for the `dialog` release with updated dependencies first

Comment (web-padawan) [reply bower.json:51]: This is a dev dependency, so I think we can keep it as it is.

File: bower.json (hunk 1 above)
Comment (web-padawan) [reply bower.json]: Thanks, done.

**Comment by web-padawan** [comment]
For some reason I'm getting `yarn install --flat` failing:

```
info Unable to find a suitable version for "@vaadin/vaadin-themable-mixin", please choose one by typing one of the numbers below:
  1) "@vaadin/vaadin-themable-mixin@^1.5.2, @vaadin/vaadin-themable-mixin@^1.5.2, @vaadin/vaadin-themable-mixin@^1.5.2, @vaadin/vaadin-themable-mixin@^1.5.2, @vaadin/vaadin-themable-mixin@^1.5.2, @vaadin/vaadin-themable-mixin@^1.5.2, @vaadin/vaadin-themable-mixin@^1.5.2" which resolved to "1.5.2"
  2) "@vaadin/vaadin-themable-mixin@^1.2.1, @vaadin/vaadin-themable-mixin@^1.2.1, @vaadin/vaadin-themable-mixin@^1.2.1, @vaadin/vaadin-themable-mixin@^1.2.1, @vaadin/vaadin-themable-mixin@^1.2.1, @vaadin/vaadin-themable-mixin@^1.3.2, @vaadin/vaadin-themable-mixin@^1.2.1" which resolved to "1.4.0"
```

Wondering why `@^1.2.1" which resolved to "1.4.0"` happens. It should resolve properly.

**Comment by web-padawan** [comment]
The problem with resolutions was apparently caused by overlay dist-tag `latest` set to 3.3.0

Fixed by changing it to be 3.4.0:

```
$ npm dist-tag add @vaadin/vaadin-overlay@3.4.0 latest
+latest: @vaadin/vaadin-overlay@3.4.0
```

**Pull request merged**